src/modules/components.js

```diff
@@ -61,19 +65,107 @@ class Components {
     popUpDiv.appendChild(title);
     popUpDiv.appendChild(summary);
     popUpDiv.appendChild(movieInfo);
+    popUpDiv.appendChild(this.createReservarionFrom());
     app.appendChild(popUpDiv);
+    this.paintReservations(movie.id);
 
     mainContent.classList.toggle('contient-in-backgound');
     return app;
   };
 
+  createReservarionFrom = () => {
+    const reservationFormDiv = document.createElement('div');
+    reservationFormDiv.className = 'reservation-form-div';
+
+    const formTitle = document.createElement('h1');
+    formTitle.textContent = 'Add a reservation';
+
+    const form = document.createElement('form');
+    form.className = 'reservation-form';
+
+    const nameInput = document.createElement('input');
+    nameInput.className = 'name-input';
+    nameInput.placeholder = 'Your name';
+    nameInput.type = 'text';
+    nameInput.setAttribute('required', true);
+
+    const dateInput = document.createElement('input');
+    dateInput.className = 'date-input start-date';
+    dateInput.placeholder = 'Start Date';
+    dateInput.type = 'date';
+    dateInput.setAttribute('required', true);
+
+    const dateInputEnd = document.createElement('input');
+    dateInputEnd.className = 'date-input end-date';
+    dateInputEnd.placeholder = 'End Date';
+    dateInputEnd.type = 'date';
+    dateInputEnd.setAttribute('required', true);
+
+    const submitButton = document.createElement('button');
+    submitButton.className = 'submit-input';
+    submitButton.textContent = 'Submit';
+
+    form.appendChild(nameInput);
+    form.appendChild(dateInput);
+    form.appendChild(dateInputEnd);
+    form.appendChild(submitButton);
+    reservationFormDiv.appendChild(formTitle);
+    reservationFormDiv.appendChild(form);
+    return reservationFormDiv;
+  };
+
   closeReservationPopUp = () => {
     const mainContent = document.querySelector('.movies-contient');
     const popUpDiv = document.querySelector('.reservation-div');
 
     mainContent.classList.toggle('contient-in-backgound');
     popUpDiv.remove();
   };
+
+  reservations = (reservationArray = null) => {
+    const div = document.createElement('div');
+    div.className = 'reservations-div';
+    const title = document.createElement('h2');
+    title.className = 'reservation-title';
+    title.innerHTML = `Reservations (${this.getReservationCount(reservationArray)})`;
+
+    div.appendChild(title);
+    if (reservationArray.error) {
+      return div;
+    }
+
+    if (document.querySelector('.reservations-div')) {
+      document.querySelector('.reservations-div').remove();
+    }
+
+    const ul = document.createElement('ul');
+    reservationArray.reverse().forEach((reservation) => {
+      const li = document.createElement('li');
+      li.textContent = `${reservation.date_start} - ${reservation.date_end} by ${reservation.username}`;
+      ul.appendChild(li);
+    });
+    div.appendChild(ul);
```

Comment:
In the Browser, this doesnt quite show as intended, as it shows only the start date for both dates as seen in the screenshot below

![Uploading image.png…]()

Reply:
Thanks for spotting this. I have fixed the issue and pushed the changes.
Thanks for the feedback 👍